autoload/ctrlsf/buf.vim

```diff
@@ -50,17 +50,23 @@ func! ctrlsf#buf#SetLine(buf_name, lnum, content) abort
 
         let l:buf_num = bufnr(a:buf_name)
 
-        " setbufline will append if a:num is the line after the last.
-        " nvim_buf_set_lines needs different arguments to do so.
-        if a:lnum == 1 + nvim_buf_line_count(l:buf_num)
-            let l:lfrom = a:lnum - 1
-            let l:lto = l:lfrom
+        " Try to mimic setbufline's behavior
```

Comment:
Taking these lines into a function like `ctrlsf#buf#NeoPseudoSetBufLine` may be more clear.